src/test/java/org/kie/processmigration/service/SchedulerServiceImplTest.java

```diff
@@ -0,0 +1,212 @@
+        // Given
+        MigrationDefinition definition = new MigrationDefinition();
+        definition.setPlanId(1L);
+        definition.setKieServerId("kie-server-1");
+        definition.setRequester("me");
+        definition.setProcessInstanceIds(List.of(1L));
+        Instant when = Instant.now().plus(500, ChronoUnit.MILLIS);
+        definition.setExecution(new Execution().setType(Execution.ExecutionType.ASYNC).setScheduledStartTime(when));
+        Migration migration = new Migration(definition);
+        migration.id = 99L;
+        CountDownLatch count = new CountDownLatch(1);
+        scheduler.getListenerManager().addJobListener(new TestJobListener(count), allJobs());
+        when(migrationService.get(migration.id)).thenReturn(migration);
+
+        // When
+        schedulerService.scheduleMigration(migration);
+
+        // Then
+        assertThat(scheduler.checkExists(new JobKey(migration.id.toString())), is(Boolean.TRUE));
+        assertThat(scheduler.checkExists(new TriggerKey(migration.id.toString())), is(Boolean.TRUE));
+        count.await(10, TimeUnit.SECONDS);
+
+        verify(migrationService, times(1)).get(99L);
+        verify(migrationService, times(1)).migrate(migration);
+    }
+
+
+    @Test
+    void testReScheduleMigration() throws InvalidMigrationException, InterruptedException, MigrationNotFoundException, SchedulerException {
+        // Given
+        MigrationDefinition definition = new MigrationDefinition();
+        definition.setPlanId(1L);
+        definition.setKieServerId("kie-server-1");
+        definition.setRequester("me");
+        definition.setProcessInstanceIds(List.of(1L));
+        Instant when = Instant.now().plus(5, ChronoUnit.HOURS);
+        definition.setExecution(new Execution().setType(Execution.ExecutionType.ASYNC).setScheduledStartTime(when));
+        Migration migration = new Migration(definition);
+        migration.id = 99L;
+        CountDownLatch count = new CountDownLatch(1);
+        scheduler.getListenerManager().addJobListener(new TestJobListener(count), allJobs());
+        when(migrationService.get(migration.id)).thenReturn(migration);
+
+        schedulerService.scheduleMigration(migration);
+
+        // When
+        migration.getDefinition().getExecution().setScheduledStartTime(Instant.now().plus(1, ChronoUnit.SECONDS));
+        schedulerService.reScheduleMigration(migration);
+
+        // Then
+        assertThat(scheduler.checkExists(new JobKey(migration.id.toString())), is(Boolean.TRUE));
+        assertThat(scheduler.checkExists(new TriggerKey(migration.id.toString())), is(Boolean.TRUE));
+        count.await(10, TimeUnit.SECONDS);
+
+        verify(migrationService, times(1)).get(99L);
+        verify(migrationService, times(1)).migrate(migration);
+    }
+
+
+    @Test
+    void testScheduleMigrationNotFound() throws InvalidMigrationException, InterruptedException, MigrationNotFoundException, SchedulerException {
```

Comment:
checked exception `MigrationNotFoundException` is not thrown in this method
```suggestion
    void testScheduleMigrationNotFound() throws InvalidMigrationException, InterruptedException, SchedulerException {
```